src/iqm/cirq_iqm/iqm_sampler.py

```diff
@@ -65,12 +66,20 @@ def __init__(
         **user_auth_args,  # contains keyword args auth_server_url, username and password
     ):
         self._client = IQMClient(url, client_signature=f'cirq-iqm {version("cirq-iqm")}', **user_auth_args)
+        self._use_default_calibration_set = False
         if device is None:
-            device_metadata = IQMDeviceMetadata.from_architecture(self._client.get_quantum_architecture())
+            device_metadata = IQMDeviceMetadata.from_architecture(
+                self._client.get_dynamic_quantum_architecture(calibration_set_id)
+            )
             self._device = IQMDevice(device_metadata)
+            if calibration_set_id is None:
+                self._use_default_calibration_set = True
         else:
             self._device = device
-        self._calibration_set_id = calibration_set_id
+        if self._device.metadata.architecture is None:
+            self._calibration_set_id = calibration_set_id
+        else:
+            self._calibration_set_id = self._device.metadata.architecture.calibration_set_id
```

Comment:
I would suggest hiding `device` behind the `*` so the user has to be explicit if they want to provide it.

Reply:
What should happen if user gives both device and calibration_set_id?

Reply:
In that case, wouldn't you run into trouble if the device does not match the calibration_set_id information? Isn't it better to check and throw an error immediately?

Reply:
True, if the given device is not compatible with the calset, submitting circuits would fail. I made now some changes to `IQMSampler.__init__`, so that now it checks device compatibility with the calset if `device` is explicitly passed.